thoth/lab/inspection.py

```diff
@@ -808,3 +808,403 @@ def create_plot_from_df(
+    df["job_log__stdout__@result__elapsed"] = df["job_log__stdout__@result__elapsed"]/1000
+    s_plot = df.plot.scatter(x='job_log__stdout__@result__elapsed', y='index', c='DarkBlue',
+                             title='Elapsed Duration', ax=ax3)
+    s_plot.set_xlabel("elapsed [ms]")
+
+    # Plot lag plot of job duration sorted by the start of job
+    df = df.sort_values(by=["status__job__started_at"])
+    lag_plot(df["status__job__duration"], ax=ax4)
+    ax4.set_title("Job Duration Autocorrelation (Sort by job_started)")
+
+    # Plot lag plot of lead time sorted by job created
+    df = df.sort_values(by=["created"])
+    lag_plot(df["lead_time"], ax=ax5)
+    ax5.set_title("lead_duration Autocorrelation (Sort by created)")
+
+    # Plot lag plot of lead time sorted by job started
+    df = df.sort_values(by=["status__job__started_at"])
+    lag_plot(df["lead_time"], ax=ax6)
+    ax6.set_title("lead_duration Autocorrelation (Sort by job_started)")
+
+
+def plot_subcategories_by_hues(df_cat: pd.DataFrame, df: pd.DataFrame, column):
+    """Create scatter plots with parameter categories separated by hues.
+
+    :param df_cat: filtered dataframe with columns to analyze as returned by columns_to_analyze
+    :param df: data frame with duration information as returned by process_inspection_results
+    :param colum: job duration/build duration columns from 'df'
+    """
+    df = df.reset_index()
+    for i in df_cat:
+        g = sns.FacetGrid(df, hue=i, margin_titles=True, height=5, aspect=1)
+        g.map(sns.regplot, column, "index", fit_reg=False, x_jitter=.1,  scatter_kws=dict(alpha=0.5))
+        g.add_legend()
+
+
+def concatenated_df(lst_of_df, column):
+    """Reorganize dataframe to show the distribution of jobs in a category across different subsets of data.
+
+    :param lst_of_df: inspection result dataframes which can be different datasets or subset of datasets
+    :param column: category for grouping to see the distribution of results
+    """
+    lst_processed = []
+    for i in lst_of_df:
+        i = i.reset_index()
+        df_grouping_category = i.groupby([column]).count()
+        col_one = df_grouping_category["index"]
+        df_col_one = pd.DataFrame(col_one)
+        df_col_one = df_col_one.rename(index=str, columns={"index": "Total jobs:"+"{}".format(len(i))})
+        lst_processed.append(df_col_one)
+
+    df_final = pd.concat([i for i in lst_processed], axis=1)
+    return df_final
+
+
+def summary_trace_plot(df: pd.DataFrame, df_categories: pd.DataFrame, lst=[]):
+    """Create trace plot scaled by percentage of compositions of each parameter analyzed.
+
+    :param df: data frame with duration information as returned by process_inspection_results
+    :param df_categories: filtered dataframe with columns to analyze as returned by columns_to_analyze
+    :param lst: list of set or subset dataframes of data with the last value in the list being entire data set
```

Comment:
This parameter is extremely confusing to me. Perhaps a type hint with `typings` would help at least a bit?